cylc/flow/scheduler.py

```diff
@@ -1771,6 +1775,29 @@ def _get_events_conf(self, key, default=None):
         return self.suite_event_handler.get_events_conf(
             self.config, key, default)
 
+    def process_cycle_point_opts(self) -> None:
+        """Check the values of --icp, --fcp, --startcp, --stopcp.
+
+        Reset the values to None if necessary:
+        * The value 'ignore' is not used in a first start.
+        * The opts --icp and --startcp cannot be used in a restart.
+        """
```

Comment:
I'm certain that `--startcp` can't be used in a restart. And, previously with `cylc restart`, there was no `--icp` option for changing the icp. However, you could in fact use `--ignore-initial-cycle-point` with `cylc restart`, which allows you to change the icp. 

So, should we
1. Allow the use of `--icp=CYCLE_POINT` for a restart?
2. Or disallow the use of `--icp=ignore` for a restart?